src/ipc/handler.c

```diff
@@ -665,11 +665,19 @@ static int ipc_pm_gate(uint32_t header)
 
 	IPC_COPY_CMD(pm_gate, ipc_get()->comp_data);
 
+	/* pause dma trace firstly if needed */
+	if (pm_gate.flags & SOF_PM_NO_TRACE)
+		trace_off();
```

Comment:
Do you flush trace buffer here ?

Reply:
@lgirdwood I am afraid there is no easy way to flush the buffer atm due to the current implementation of the buffer copying to the host. The flush done upon entering panic mode is something different. Once the trace mechanics is simplified (its working thread job vs. things done synchronously), the flush will be easy to add. Atm we could live with that imo esp. that not too much activity is expected right before entering D0ix.

Reply:
@mmaka1 ok, so I guess we get this data once we re-enable trace.